vmm-reference-simple-handler.md

```diff
@@ -0,0 +1,137 @@
+# [Simple Handler](./../src/devices/src/virtio/net/simple_handler.rs)
+
+`SimpleHandler` contains methods for handling Tx/Rx queues. This handler is generalized over the implementaton of queues (does not make any assumptions the way queues are implemented). 
+
+```rs
+pub struct SimpleHandler<M: GuestAddressSpace, S: SignalUsedQueue> {
+    pub driver_notify: S,
+    pub rxq: Queue<M>,
+    pub rxbuf_current: usize,
+    pub rxbuf: [u8; MAX_BUFFER_SIZE],
+    pub txq: Queue<M>,
+    pub txbuf: [u8; MAX_BUFFER_SIZE],
+    pub tap: Tap,
+}
+
+```
+
+__A brief overview of all the struct variables:__
+
+1. `driver_notify:`
+    
+    A SignalUsedQueue object to model the operation of signalling the driver about used events. It should implement a method named `signal_used_queue` that takes parameter `index` and notifies the driver about index. Index values used here in SimpleHandler are `RXQ_INDEX` (for receiving) and `TXQ_INDEX` (for transmitting).
+
+2. `rxq:`  Recieving queue of `GuestAddressSpace`
+3. `rxbuf_current:` Number of bytes in `rxbuf` 
+4. `rxbuf:` Recieving buffer of maximum size `MAX_BUFFER_SIZE`
+5. `txq:`  Transmit queue of `GuestAddressSpace`
+6. `txbuf:` Transmit buffer of maximum size `MAX_BUFFER_SIZE`
+7. `tap:` Object of Tap struct that wraps the file descriptor for the tap device so methods can run ioctls on the interface
+
+```rs
+const MAX_BUFFER_SIZE: usize = 65562;
+```
+<table>
+<caption>
+Incomping Packet Size
+</caption>
+    <tr>
+        <th colspan="2">
+        &nbsp;
+        vitio
+        &nbsp;&nbsp;&nbsp;
+        &nbsp;&nbsp;&nbsp;&nbsp;
+        ethernet header + UDP/TCP Packet
+        </th>
+    </tr>
+    <tr>
+        <td style="outline: thin solid">
+        </td>
+        <td style="outline: thin solid">
+          &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
+          &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
+          &nbsp;&nbsp;&nbsp;&nbsp;
+        </td>
+    </tr>
+    <tr>
+        <td colspan="2">
+        12 bits
+        &nbsp;&nbsp;&nbsp;&nbsp;&thinsp;
+        &nbsp;&nbsp;&nbsp;&nbsp;&thinsp;
+        &nbsp;&nbsp;&nbsp;&nbsp;&thinsp;
+        &nbsp;&nbsp;&nbsp;&nbsp;&thinsp;
+        &nbsp;&nbsp;&nbsp;&nbsp;&thinsp;
+        65550 bits
+    <br>
+        <div align = "center">65562 bits</dev>
+      </td>
+    </tr>
+</table>
+
+__A brief overview of all the struct methods:__
+
+1. `Constructor:`
+
+```rs
+pub fn new(driver_notify: S, rxq: Queue<M>, txq: Queue<M>, tap: Tap) -> Self
+```
+Creates a SimpleHandler with provided driver_notify, rxq, txq and tap. Sets rxbuf_current and all entries of Tx and Rx buffers to 0.
```

Comment:
Creates a SimpleHandler with the provided driver_notify, rxq, txq, and tap. Sets rxbuf_current and all entries of the Tx and Rx buffers to 0.